assignment/Main.hs

```diff
@@ -5,6 +5,10 @@ import Graphics.Gloss
 -- forward pipeline
 (|>) x f = f x
 
+--I'm not entirely sure how to do parts 3 or 4. Below is my best guess, but I'm not sure how to interpret the errors. Apologies--I will try to start earlier
+--and set up a meeting if I get stuck next time. 
+format seconds = ((String)(seconds / 60) ++ ":" ++ (String)(seconds mod 60))
```

Comment:
There are a couple of problems in this line.

First, I think you're trying to do a type cast with `(String)`. Haskell doesn't do casting, because it is strongly typed. Instead, we convert the number to a string using `show (seconds mod 60)`. But, if you do that, you'll get another error because `mod` is used incorrectly. Firstly, it's a function, so you need to write `mod seconds 60` (or put `mod` in backticks to turn it into an operator). But, secondly, `mod` is only defined to work in integers, so this would still not work.

One way around this is to convert the incoming value to an integer:

~~~ haskell
format seconds = 
  show (div sec 60)  ++ ":" ++ show (rem sec 60)
  where
    sec = floor seconds
~~~